### ISSUE TITLE
[BUG]: Problem with NER: Function __gather_flair_tokens(...) sets wrong context

### ISSUE THREAD

**Issue by kobiche** [flairNLP/flair#3345]
**Context**

Suppose we have the following Flair-Sentence list with the NER entities _date_, _last_name_, and _first_name_
```
sentence = [Sentence: "Date: Tuesday, 05 .09.2045 15:33" → ["05 .09.2045"/date],
 Sentence: "Mike Thompson" → ["Mike"/first_name, "Thompson"/last_name],
 Sentence: "From: mike@my-cool-email.ag" → ["mike@my-cool-email.ag"/mail_address],
 Sentence: "To: Administration"]
```

The sentences come from the same document and are in almost the same order as in the list. The list is actually ordered by sentence length (in reality the order is [1, 2, 0, 3]). All sentences have their context set (i.e. `sentence.previous_sentence is not None`), but the second one. The second sentence _Mike Thompson_ is the first one in the document and has as `next_sentence()` _From: mike@my-cool-email.ag__ and as `previous_sentence()` _None_. Therefore the method `is_context_set()` returns _False_.

**Problem**

I think the function 
https://github.com/flairNLP/flair/blob/42ea3f6854eba04387c38045f160c18bdaac07dc/flair/embeddings/transformer.py#L641

has 2 problems:
1. it overwrites the next_sentence of _Mike Thompson_ with None. This leaves the sentence without context! (Remember, it has no previous context because it is the first sentence in the document)
2. it only makes sense to set the context here if the list of sentences is ordered; the function has no parameter to control this

Therefore, I suggest adding a parameter to control context generation, with a warning that the context will be created after the given list and may overwrite the existing context.

**Comment by helpmefindaname** [comment]
Hi @kobiche 
can you check if that is still a problem on the current master branch? I think that should be already fixed.

If yes, can you please create a bug report, using the BUG-Issue template?

**Comment by kobiche** [comment]
You are right. It is already solved in master. Probably I was working with an old version. 
Thanks for your response!

**Comment by alanakbik** [comment]
Master was also just released as Flair 0.13.0, i.e. you can install this version through pip :)
